server/game/cards/02_SHD/units/CriminalMuscle.ts

```diff
@@ -0,0 +1,24 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+
+export default class CriminalMuscle extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '6475868209',
+            internalName: 'criminal-muscle'
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addWhenPlayedAbility({
+            title: 'Return a non-unique upgrade to its owner\'s hand.',
+            optional: true,
+            targetResolver: {
+                cardCondition: (card) => card.isUpgrade() && !card.unique,
```

Comment:
upgrade should be on zonefilter

Reply:
@VJubert I don't see `Upgrade` in the `ZoneName` or `WildcardZoneName` used on `ZoneFilter`. Should I use a `zoneFilter: WildcardZoneName.AnyArena` or are you saying I should add a new zone?

Reply:
I'm sorry it's not on zoneFilter but on cardTypeFilter : WildcardCardType.Upgrade

Reply:
@VJubert would you recommend:

```typescript
  targetResolver: {
      cardCondition: (card) => card.isUpgrade() && !card.unique,
      cardTypeFilter: WildcardCardType.Upgrade,
      immediateEffect: AbilityHelper.immediateEffects.returnToHand()
  }
```

or

```typescript
  targetResolver: {
      cardCondition: (card) => !card.unique,
      cardTypeFilter: WildcardCardType.Upgrade,
      immediateEffect: AbilityHelper.immediateEffects.returnToHand()
  }
```

I am thinking `card.isUpgrade()` is redundant with the `cardTypeFilter`. Is that the case?

Reply:
Yeah they're redundant, the filter is preferred if possible

Reply:
Updated.